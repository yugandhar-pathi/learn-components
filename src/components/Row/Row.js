import PropTypes from 'prop-types'
import {RowStyle} from './Row.styles'

export const Row = ({justifyContent, dataTestId, children}) => {
    
    return(
        <RowStyle data-testid={dataTestId} justifyContent={justifyContent}>{children}</RowStyle>
    )
}

Row.propTypes = {
    justifyContent: PropTypes.oneOf(["space-between", "space-around", "center"]),
    children :  PropTypes.arrayOf(PropTypes.element).isRequired,
    dataTestId : PropTypes.string.isRequired
}

Row.defaultProps = {
    justifyContent : "space-around"
}
