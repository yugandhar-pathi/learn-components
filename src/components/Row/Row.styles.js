import styled from 'styled-components'

export const RowStyle = styled.div`
        display: flex;
        flex-direction: row;
        justify-content: ${(props)=> props.justifyContent}
`