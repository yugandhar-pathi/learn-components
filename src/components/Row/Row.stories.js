import React from 'react';
import {Row} from './Row'

export default{
    component : Row,
}

export const spaceBetween = () => <Row justifyContent="space-between"><p>Hi</p><p>Hi</p></Row>
export const spaceAround = () => <Row justifyContent="space-around"><p>hello</p><p>hello</p></Row>
export const center = () => <Row justifyContent="center"><p>Hello World</p></Row>