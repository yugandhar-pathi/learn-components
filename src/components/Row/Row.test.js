import {render, screen} from '@testing-library/react';
import {Row} from './Row';

test("Row Component", () =>{
    render(<Row dataTestId="row-id">
     <p>hi</p>
     <p>Hello</p> 
    </Row>)
    expect(screen.getByTestId("row-id").textContent).toBe("hiHello")
})

test("Snapshot for row", ()=>{
    render(<Row dataTestId="row-id">
     <p>hi</p>
     <p>Hello</p>
    </Row>)
    expect(screen.getByTestId("row-id")).toMatchSnapshot()
})
