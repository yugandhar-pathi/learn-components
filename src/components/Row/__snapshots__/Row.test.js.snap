// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for row 1`] = `
<div
  class="sc-bczRLJ XhHPr"
  data-testid="row-id"
    <p>
      hi
    </p>
    <p>
      Hello
    </p>
</div>
`;
